src/distance/ConfigurationRanker.cpp

```diff
@@ -0,0 +1,64 @@
+#include "aikido/distance/ConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+ConfigurationRanker::ConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton,
+    const std::vector<statespace::StateSpace::State*> ikSolutions)
+  : mMetaSkeletonStateSpace(std::move(metaSkeletonStateSpace))
+  , mMetaSkeleton(std::move(metaSkeleton))
+{
+  if (!mMetaSkeletonStateSpace)
+    throw std::invalid_argument("MetaSkeletonStateSpace is nullptr.");
+
+  if (!mMetaSkeleton)
+    throw std::invalid_argument("MetaSkeleton is nullptr.");
+
+  if (ikSolutions.empty())
+    throw std::invalid_argument("Vector of IK Solutions is empty.");
+
+  mIKSolutions.resize(ikSolutions.size());
+  mIKSolutions.shrink_to_fit();
```

Comment:
I think not having `mIKSolutions` allows us to skip all this initialization code.